test/lib/jdk/test/lib/jvmti/jvmti_common.h

```diff
@@ -880,6 +881,27 @@ set_event_notification_mode(jvmtiEnv* jvmti, JNIEnv* jni, jvmtiEventMode mode, j
   check_jvmti_status(jni, err, "jvmti_common set_event_notification_mode: Error in JVMTI SetEventNotificationMode");
 }
 
+static jvmtiError
+set_ext_event_notification_mode(jvmtiEnv* jvmti, jvmtiEventMode mode, const char* ename, jthread event_thread) {
+  jvmtiError err = JVMTI_ERROR_NONE;
+  jvmtiExtensionEventInfo* info = find_ext_event(jvmti, ename);
+
+  if (NULL == info) {
+    LOG("jvmti_common set_ext_event_callback: Extension event was not found: %s\n", ename);
+    err = JVMTI_ERROR_NOT_AVAILABLE;
+  } else {
+    err = jvmti->SetEventNotificationMode(mode, (jvmtiEvent)info->extension_event_index, event_thread);
+  }
+
+  return err;
+}
+
+static void
+set_ext_event_notification_mode(jvmtiEnv* jvmti, JNIEnv* jni, jvmtiEventMode mode, const char* ename, jthread event_thread) {
```

Comment:
I don't see anything that suggests it's safe to assume this is only used in a C++ context (and the extension is not `.hpp`), in which case it's not reasonable (nor necessary) to overload the name `set_ext_event_notification_mode`.

Also, the code should be written in the style of the surrounding code:
```c
static jvmtiError
set_ext_event_notification_mode(jvmtiEnv* jvmti, jvmtiEventMode mode, const char* ename, jthread event_thread) {
  jvmtiExtensionEventInfo* info = find_ext_event(jvmti, ename);
  if (NULL == info) {
    LOG("jvmti_common set_ext_event_callback: Extension event was not found: %s\n", ename);
    return JVMTI_ERROR_NOT_AVAILABLE;
  }
  return jvmti->SetEventNotificationMode(mode, (jvmtiEvent)info->extension_event_index, event_thread);
}
```

Reply:
In JDK-next, the compiler used to build `jvmti_common.h` should support function overloading in C. Also, there are existing instances of function overloading here: `set_event_notification_mode`. Should we still be worried about overloading?

Reply:
Removed 5-argument version.